src/compiler/core.ts

```diff
@@ -1492,6 +1492,47 @@ namespace ts {
         return createMultiMap() as UnderscoreEscapedMultiMap<T>;
     }
 
+    export function createQueue<T>(items?: readonly T[]): Queue<T> {
+        const elements: (T | undefined)[] = items?.slice() || [];
+        let headIndex = 0;
+
+        function isEmpty() {
+            return headIndex === elements.length;
+        }
+
+        function enqueue(...items: T[]) {

```

Comment:
Would have rather just limited it to one element.

Reply:
Wait, even if I eliminate `enqueue`'s spread, the items will still need to be spread into `push`.  Maybe you're saying there should be a foreach at the call site?

Reply:
This only happens in FAR and I directly replaces `push(...foo)` with `enqueue(...foo)`, so it shouldn't cause any regression.  It might be wise to keep an eye out for crashes from that code though.